openedx/core/djangoapps/discussions/api/external_discussions.py

```diff
@@ -0,0 +1,73 @@
+from django.db import IntegrityError
+from opaque_keys.edx.keys import LearningContextKey, UsageKey
+
+from .data import ExternalDiscussionData
+from ..models import ExternalDiscussionsIdMapping
+
+
+def create_external_discussion_mapping(
```

Comment:
We don't appear to have a helper to update existing mappings; should we?

Reply:
As far as I can tell, the scenarios in which this would be needed are:

* The external service changes the id they use to refer to a specific discussion
* We transplant an existing discussion from one context to another within an edX deployment
* We want to include a new discussion either from the same platform, or a new platform

> The external service changes the id they use to refer to a specific discussion

This seems very unlikely to me

> We transplant an existing discussion from one context to another within an edX deployment

This also seems unlikely given the discussion would be very out of context if it were to be transplanted

> We want to include a new discussion either from the same platform, or a new platform

Realistically, this seems like the only possibility we might want to support. Currently though, this is not possible given the model constraints, as the `usage_key` and `context_key` together form a `UniqueConstraint`. In order to support this, we would need to remove this constraint and have `get_external_discussion_context` return only the most recent/appropriate mapping

CC @xitij2000, as you might want to weigh in on this